src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
```

Comment:
Consider renaming `currentState` to `stateCopy` to adhere to the checklist's naming conventions for object copies.